langs/go/gowrapper.c

```diff
@@ -0,0 +1,113 @@
+#include <dirent.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#define ERR_AND_EXIT(msg) do { perror(msg); exit(EXIT_FAILURE); } while (0)
+
+void copy_file(const char* src_file, const char* dst_file);
+void copy_folder(const char* src_dir, const char* dst_dir);
+
+const char* go = "/usr/local/bin/go", *code = "code.go";
+
+int main(int argc, char* argv[]) {
+    if (!strcmp(argv[1], "--version")) {
+        execl(go, go, "version", NULL);
+        ERR_AND_EXIT("execl");
+    }
+
+    // Copy the cache to /tmp so it's mutable, this is hacky but faster than
+    // running Go without the stdlib cached.
+    copy_folder("/cache", "/tmp/cache");
```

Comment:
I don't think I agree on this operation being hacky, really. This seems comparable to a scenario where assets are first-time downloaded during a compilation process, and this copy is just that _download_ putting everything in its or one of its lookup path(s), kind of what's done for Gleam and what's about to be done for Scala's very latest.